azurerm/internal/services/servicefabricmesh/resource_arm_service_fabric_mesh_application.go

```diff
@@ -0,0 +1,436 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmServiceFabricMeshApplication() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmServiceFabricMeshApplicationCreateUpdate,
+		Read:   resourceArmServiceFabricMeshApplicationRead,
+		Update: resourceArmServiceFabricMeshApplicationCreateUpdate,
+		Delete: resourceArmServiceFabricMeshApplicationDelete,
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.ServiceFabricMeshApplicationID(id)
+			return err
+		}),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(90 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(90 * time.Minute),
+			Delete: schema.DefaultTimeout(90 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			// Follow casing issue here https://github.com/Azure/azure-rest-api-specs/issues/9330
+			"resource_group_name": azure.SchemaResourceGroupNameDiffSuppress(),
+
+			"location": azure.SchemaLocation(),
+
+			"service": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+						"os_type": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(servicefabricmesh.Linux),
+								string(servicefabricmesh.Windows),
+							}, false),
+						},
+						"code_package": {
+							Type:     schema.TypeSet,
+							Required: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"name": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validation.StringIsNotEmpty,
+									},
+									"image_name": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validation.StringIsNotEmpty,
+									},
+									"resources": {
+										Type:     schema.TypeList,
+										Required: true,
+										MaxItems: 1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"requests": {
+													Type:     schema.TypeList,
+													Required: true,
+													MaxItems: 1,
+													Elem: &schema.Resource{
+														Schema: map[string]*schema.Schema{
+															"memory": {
+																Type:         schema.TypeFloat,
+																Required:     true,
+																ValidateFunc: validation.FloatAtLeast(0),
```

Comment:
Is there no maximum value for these propeties?

Reply:
I can't find a definitive answer. The error message that's returned doesn't seem helpful for determining what the max limit is on any of the values where I don't specify a max. In the same error message we get that the max value is 14 or 42. 

```
{"code":"MemoryPerCodePackageExceededQuota","message":"Request exceeds available quota for memory per code package. Service:testservice1. CodePackage:testcodepackage1. Quota: 14. Requested: 10000000000."},{"code":"MemoryPerReplicaExceededQuota","message":"Request exceeds available quota for memory per service replica. Service:testservice1. Quota: 14. Requested: 10000000000."},{"code":"MemoryPerApplicationExceededQuota","message":"Request exceeds available quota for memory per application. Quota: 42. Requested: 10000000000."},{"code":"Exception","message":"Microsoft.SeaBreeze.SubscriptionManager.Quota.QuotaValidationException: One or more quotas were exceeded by current request.
```

Unfortunately, I think the best course of action is to leave it up to the user to know what's best for their service.